components/html-2-objects-compare.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * File:  html-2-objects-compare.js
+ *
+ * This component generates an HTML file that shows a deep dump of two Javascript objects side by side
+ * for easy visual comparison.
+ *
+ * Data coming into the "left" port will be shown in the left side column; data coming into the "right"
+ * port will be shown in the right side column of the HTML page.  The title will be taken from the IIP 
+ * constant configuration for each port in the form of {"title":"<column title>"}.
+ */
+ 
+var _ = require('underscore');
+var util = require('util');
+var fs = require('fs');
+var open = require('open');
+
+var jswrapper = require('../src/javascript-wrapper');
+
+// Count number of executions.  We don't want to automatically open the comparison file the
+// firt time this component executes since it would inhibit viewing the noflo-graph.  Wait until
+// second (manually executed)  execution before opening the html comparision file.
+var executionCount = 0;
+
+exports.getComponent = jswrapper({
+
+    description: "Generate an HTML page that compares two JavaScript objects side-by-side for easy visual comparison.",
+
+    inPorts: { 
+        file: {
+            datatype: 'string',
+            description: "path to the html file to be created.",
+            addressable: true,
+            required: true
+        },
+        left: { 
+            datatype: 'object',
+            description: "a JavaScript object to display on the left side side of the HTML page.",
+            addressable: true,
+            required: true
+        },
+        right: { 
+            datatype: 'object',
+            description: "a JavaScript object to display on the right side side of the HTML page.",
+            addressable: true,
+            required: true
+        }
+    },
+
+    /**
+     * This updater will generate a simple HTML page that visually displays the content of 
+     * both the left and right objects.
+     *
+     * @param file file path to which the html should be written
+     * @param left Javascript object to display on left side of HTML page and an IIP 
+     *             setting of format {"title":"<title>"}
+     * @param right Javascript object to display on right side of HTML page and an IIP 
+     *             setting of format {"title":"<title>"}
+     */
+    updater: function( file, left, right ) {   
+
+        // Verify that we have all data required to generate the html page. 
+        // TODO: Revisit and remove this check after more of the pipeline architecture is completed.
+        if ( shouldFireUpdater( file, left, right) ) { 
+
+            // We have both an IIP and port data on these ports.  It can theoretically arrive in any order.
+            // sortPortData return an object with a title and a data field so we know what we are referencing
+            // henceforth.
+            var leftOutput = sortPortData( left, 'left' );
```

Comment:
Because the title is what will be displayed as the column title of that json object.  I thought it made sense to keep related things together.  I also chose to do it this way so we had a test component that had both IIP and packet input on the same port since it's a scenario we will need to be testing in the framework.

This component will be displaying one object on the left (with the IIP title specified for that port/object,and one object on the right (with it's own IIP title).